src/rgw/driver/rados/rgw_trim_bilog.cc

```diff
@@ -253,8 +253,20 @@ class BucketTrimWatcher : public librados::WatchCtx2 {
   BucketTrimWatcher(rgw::sal::RadosStore* store, const rgw_raw_obj& obj,
                     TrimCounters::Server *counters)
     : store(store), obj(obj) {
-    handlers.emplace(NotifyTrimCounters, new TrimCounters::Handler(counters));
-    handlers.emplace(NotifyTrimComplete, new TrimComplete::Handler(counters));
+    std::unique_ptr<TrimCounters::Handler> counter_handler(
+        new TrimCounters::Handler(counters));
+    std::unique_ptr<TrimComplete::Handler> complete_handler(
+        new TrimComplete::Handler(counters));
+
+    auto result = handlers.emplace(NotifyTrimCounters, std::move(counter_handler));
```

Comment:
can we just replace the `new TrimCounters::Handler(counters)` arguments with `std::make_unique<TrimCounters::Handler>(counters)`? that should free the memory whether or not the `emplace()` succeeds